lib/Compilation.js

```diff
@@ -1609,14 +1603,15 @@ class Compilation extends Tapable {
 	}
 
 	checkConstraints() {
-		const usedIds = {};
+		const usedIds = new Set();
 
 		const modules = this.modules;
 		for(let indexModule = 0; indexModule < modules.length; indexModule++) {
 			const moduleId = modules[indexModule].id;
-
-			if(usedIds[moduleId])
+			if(typeof module !== "number") continue;
+			if(usedIds.has(moduleId))
 				throw new Error(`checkConstraints: duplicate module id ${moduleId}`);
```

Comment:
I'm pretty sure you can't have duplicate string module ids either

Reply:
You're right. I fixed the condition. Before the PR this was no-op.